### PR TITLE
Ruby/Resistor Colors: Add Mentor Notes

### PR DESCRIPTION
**Pull request by F3PiX**
We are about to merge the new side exercise Resistor Colors  https://github.com/exercism/ruby/pull/939. :tada:
I just want to get something ready for mentors.  We can adjust after we've seen some submissions.

@exercism/website-copy-ruby

### CODE REVIEW

File: tracks/ruby/exercises/resistor-colors/mentoring.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  def self.value(colors)
+    colors.map { |color| COLOR_CODES.index(color) }.join.to_i
+  end
+
+end
+
+```
+
+### General
+This is the very first Array exercise, and the first loop. Introducing `map` is key in this exercise.
```
Comment (pgaspar) [inline tracks/ruby/exercises/resistor-colors/mentoring.md]: > This is the very first Array exercise

Isn't this set up as a "bonus" exercise? ([reference](https://github.com/exercism/ruby/pull/939/commits/3ccf5c6211c57da699ea18c15d51537f5eb65663))

File: tracks/ruby/exercises/resistor-colors/mentoring.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+### Talking points 
+- _`%w` notation_: for arrays of strings. Link to [styleguide](https://github.com/rubocop-hq/ruby-style-guide#percent-w)
+- _chaining methods_ after a block (especially when people use a local variable for `join.to_i`
+- _`each` vs `map`_: this explains the difference: [each vs map](https://learn.onemonth.com/ruby-tutorial-map-vs-each/)
+- _variable vs constant_: this article explains just that: [var vs constant](https://ruby-doc.org/docs/ruby-doc-bundle/Tutorial/part_01/variables.html) 
+- _freeze_: this post has exactly the right level of information: [freeze](https://freelancing-gods.com/2017/07/27/an-introduction-to-frozen-string-literals.html); no need to dive deeper into (im)mutability.
+
+
+### Changelog
+First introduced 2019 Mar 10
```
Comment (pgaspar) [inline tracks/ruby/exercises/resistor-colors/mentoring.md]: Maybe we should have a specific format for the changelog. Maybe something like:

* Version 1: exercise first introduced on March 10, 2019

Maybe [HighScores](https://github.com/exercism/website-copy/blob/master/tracks/ruby/exercises/high-scores/mentoring.md) can be a good guide as well?

File: tracks/ruby/exercises/resistor-colors/mentoring.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+### New Concepts
+Array, Array#index, `map`, Constant, chaining methods
+
+### Minimal solution for approval
+```ruby
+
+class ResistorColors
+
+  COLOR_CODES = ["black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"]
```
Comment (SleeplessByte) [inline tracks/ruby/exercises/resistor-colors/mentoring.md]:
```suggestion
  COLOR_CODES = %w[black brown red orange yellow green blue violet grey white].freeze
```

Default style via [rubocop WordArray](https://www.rubydoc.info/github/bbatsov/RuboCop/RuboCop/Cop/Style/WordArray) and [rubocop MutableConstant](https://www.rubydoc.info/github/bbatsov/RuboCop/RuboCop/Cop/Style/MutableConstant). 

Add below this solution that:
- "Alternatives include writing out the array using brackets"; the talking points already mention this.
- `freeze` should only be a talking point but not a reason to disapprove

File: tracks/ruby/exercises/resistor-colors/mentoring.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
+Resistor Colors is a side exercise, unlocked by Hello World
+
+### New Concepts
+Array, Array#index, `map`, Constant, chaining methods
+
+### Minimal solution for approval
+```ruby
+
```
Comment (SleeplessByte) [inline tracks/ruby/exercises/resistor-colors/mentoring.md]: I suggest removing the whitespace above `class`, but adding an empty line below the header, following default [markdownlint](https://github.com/updownpress/markdown-lint/blob/master/rules/022-blanks-around-headers.md).

File: tracks/ruby/exercises/resistor-colors/mentoring.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
+Resistor Colors is a side exercise, unlocked by Hello World
+
+### New Concepts
+Array, Array#index, `map`, Constant, chaining methods
```
Comment (SleeplessByte) [inline tracks/ruby/exercises/resistor-colors/mentoring.md]: Add blank line between header and content, following default [markdownlint](https://github.com/updownpress/markdown-lint/blob/master/rules/022-blanks-around-headers.md)

File: tracks/ruby/exercises/resistor-colors/mentoring.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+class ResistorColors
+
+  COLOR_CODES = ["black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"]
+  
+  def self.value(colors)
+    colors.map { |color| COLOR_CODES.index(color) }.join.to_i
+  end
+
+end
+
```
Comment (SleeplessByte) [inline tracks/ruby/exercises/resistor-colors/mentoring.md]: Remove whitespace between final `end` and end of code block.

File: tracks/ruby/exercises/resistor-colors/mentoring.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+```
+
```
Comment (SleeplessByte) [inline tracks/ruby/exercises/resistor-colors/mentoring.md]: Add note about acceptable variations include using a `module`, `class << self`, `module_function` and the likes. We don't want mentors to comment on any of that, seeing it's such an early exercise!

**Comment by F3PiX** [comment]
Thanks @SleeplessByte and @pgaspar.  I'm merging it first, as I just want something ready for when the exercise is live. We can and will improve it later. 

🍻

**Pull request merged**